internal/gatewayapi/route.go

```diff
@@ -1433,6 +1384,56 @@ func validateDestinationSettings(destinationSettings *ir.DestinationSetting, end
 	return nil
 }
 
+func (t *Translator) processServiceImportDestinationSetting(
+	name string,
+	backendRef gwapiv1.BackendObjectReference,
+	backendNamespace string,
+	protocol ir.AppProtocol,
+	resources *resource.Resources,
+	envoyProxy *egv1a1.EnvoyProxy,
+) *ir.DestinationSetting {
+	var (
+		endpoints []*ir.DestinationEndpoint
+		addrType  *ir.DestinationAddressType
+	)
+
+	serviceImport := resources.GetServiceImport(backendNamespace, string(backendRef.Name))
+	var servicePort mcsapiv1a1.ServicePort
+	for _, port := range serviceImport.Spec.Ports {
+		if port.Port == int32(*backendRef.Port) {
+			servicePort = port
+			break
+		}
+	}
+
+	// TODO(tnareddy): Should these protocols be supported for ServiceImport?
+	// Follow-up in separate issue and PR.
+	//
+	// if servicePort.AppProtocol != nil {
+	// 	protocol = serviceAppProtocolToIRAppProtocol(*servicePort.AppProtocol)
+	// }
```

Comment:
need create an issue for this.

Reply:
https://github.com/envoyproxy/gateway/issues/5485